Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -1206,6 +1206,112 @@ String HTMLInputElement::covert_number_to_string(double input) const
+    VERIFY_NOT_REACHED();
+}
+
+// https://html.spec.whatwg.org/multipage/input.html#concept-input-step
+Optional<double> HTMLInputElement::allowed_value_step() const
+{
+    // 1. If the attribute does not apply, then there is no allowed value step.
+    if (type_state() != TypeAttributeState::Date && type_state() != TypeAttributeState::Month && type_state() != TypeAttributeState::Week && type_state() != TypeAttributeState::Time && type_state() != TypeAttributeState::LocalDateAndTime && type_state() != TypeAttributeState::Number && type_state() != TypeAttributeState::Range)
+        return {};
+
+    // 2. Otherwise, if the attribute is absent, then the allowed value step is the default step multiplied by the step scale factor.
+    auto maybe_step_string = get_attribute(AttributeNames::step);
+    if (!maybe_step_string.has_value())
+        return default_step() * step_scale_factor();
+
+    // 3. Otherwise, if the attribute's value is an ASCII case-insensitive match for the string "any", then there is no allowed value step.
+    auto step_string = maybe_step_string.value();
+    if (Infra::is_ascii_case_insensitive_match(step_string, "any"_string))
+        return {};
+
+    // 4. Otherwise, if the rules for parsing floating-point number values, when they are applied to the attribute's value, return an error, zero, or a number less than zero, then the allowed value step is the default step multiplied by the step scale factor.
+    auto maybe_step = parse_floating_point_number(step_string);
+    if (!maybe_step.has_value() || maybe_step.value() == 0 || maybe_step.value() < 0)
+        return default_step() * step_scale_factor();
+
+    // 5. Otherwise, the allowed value step is the number returned by the rules for parsing floating-point number values when they are applied to the attribute's value, multiplied by the step scale factor.
+    return maybe_step.value() * step_scale_factor();
+}
+
+// https://html.spec.whatwg.org/multipage/input.html#concept-input-min-zero
+double HTMLInputElement::step_base() const
+{
+    // 1. If the element has a min content attribute, and the result of applying the algorithm to convert a string to a number to the value of the min content attribute is not an error, then return that result.
+    auto maybe_minimum = min();
+    if (maybe_minimum.has_value())
+        return maybe_minimum.value();
+
+    // 2. If the element has a value content attribute, and the result of applying the algorithm to convert a string to a number to the value of the value content attribute is not an error, then return that result.
+    auto maybe_value = convert_string_to_number(value());
+    if (maybe_value.has_value())
+        return maybe_value.value();
+
+    // 3. If a default step base is defined for this element given its type attribute's state, then return it.
+    if (type_state() == TypeAttributeState::Week)
+        // The default step base is −259,200,000 (the start of week 1970-W01).
+        return -259200000;
```

Comment:
We consider this multi-line, thus should have `{ }` surrounding the body

Reply:
Also large numbers are easier to read with separators, at least for old dudes like me :)
```
-259'200'000
```